crates/core/src/diff/morph.rs

```diff
@@ -469,7 +469,7 @@ impl Iterator for Morph<'_> {
                             Op::Patch(Patch::PrependBefore { before: from.node }),
                             // Set newly inserted node as append parent
                             Op::Patch(Patch::PushCurrent),
-                            Op::Append {
+                            Op::AppendSiblings {
                                 from: from.clone(),
```

Comment:
The change that fixes the bug is this line, I'll remove this PR and open a PR for the test and separate one for the fix.